src/gridtools/clang/storage_dsl.hpp

```diff
@@ -24,150 +24,146 @@
+struct storage {
+  storage();
+
+  template <typename T>
+  storage(T...);
+
+  storage& operator=(const storage&);
+  storage& operator+=(const storage&);
+  storage& operator-=(const storage&);
+  storage& operator/=(const storage&);
+  storage& operator*=(const storage&);
+
+  storage& operator()(int, int, int);
+
+  /**
+   * @name 1D access
+   * @{
+   */
+  storage& operator()(direction);
+  storage& operator()(dimension);
+  storage& operator()(offset);
+  /** @} */
+
+  /**
+   * @name 2D access
+   * @{
+   */
+
+  storage& operator()(dimension, dimension);
+
+  storage& operator()(dimension, direction);
+  storage& operator()(direction, dimension);
+  storage& operator()(direction, direction);
+  /** @} */
+
+  /**
+   * @name 3D access
+   * @{
+   */
+  storage& operator()(dimension, dimension, dimension);
+
+  storage& operator()(direction, direction, direction);
+  storage& operator()(dimension, direction, direction);
+  storage& operator()(direction, dimension, direction);
+  storage& operator()(direction, direction, dimension);
+  storage& operator()(dimension, dimension, direction);
+  storage& operator()(dimension, direction, dimension);
+  storage& operator()(direction, dimension, dimension);
+  /** @} */
+
+  operator double() const;
+};
+
+struct var {
+  var();
+
+  template <typename T>
+  var(T...);
+
+  var& operator=(const var&);
+  var& operator+=(const var&);
+  var& operator-=(const var&);
+  var& operator/=(const var&);
+  var& operator*=(const var&);
+
+  /**
+   * @name 1D access
+   * @{
+   */
+  var& operator()(direction);
+  var& operator()(dimension);
+  var& operator()(offset);
+  /** @} */
+
+  /**
+   * @name 2D access
+   * @{
+   */
+  var& operator()(dimension, dimension);
+
+  var& operator()(dimension, direction);
+  var& operator()(direction, dimension);
+  var& operator()(direction, direction);
+  /** @} */
+
+  /**
+   * @name 3D access
+   * @{
+   */
+  var& operator()(dimension, dimension, dimension);
+
+  var& operator()(direction, direction, direction);
+  var& operator()(dimension, direction, direction);
+  var& operator()(direction, dimension, direction);
+  var& operator()(direction, direction, dimension);
+  var& operator()(dimension, dimension, direction);
+  var& operator()(dimension, direction, dimension);
+  var& operator()(direction, dimension, dimension);
+  /** @} */
+
+  operator double() const;
+};
 
 #ifndef GRIDTOOLS_CLANG_META_DATA_T_DEFINED
-        struct meta_data {
-            template < typename T >
-            meta_data(T...);
-        };
-        using meta_data_t = meta_data;
-        using meta_data_ijk_t = meta_data;
-        using meta_data_ij_t = meta_data;
-        using meta_data_j_t = meta_data;
-        using meta_data_j_t = meta_data;
-        using meta_data_k_t = meta_data;
-        using meta_data_scalar_t = meta_data;
+struct meta_data {
+  template <typename T>
+  meta_data(T...);
+};
+using meta_data_t = meta_data;
+using meta_data_ijk_t = meta_data;

```

Comment:
following the same argument, this is GT, so this should go into a separate file and under namespace gridtools
But again not in this PR